src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -212,7 +212,28 @@ public abstract class DetectPreviewFeatureAnalyzer : DiagnosticAnalyzer
 
         protected abstract SyntaxNode? GetPreviewInterfaceNodeForTypeImplementingPreviewInterface(ISymbol typeSymbol, ISymbol previewInterfaceSymbol);
 
-        protected abstract SyntaxNode? GetConstraintSyntaxNodeForTypeConstrainedByPreviewTypes(ISymbol typeOrMethodSymbol, ISymbol previewInterfaceConstraintSymbol);
+        private SyntaxNode? GetConstraintSyntaxNodeForTypeConstrainedByPreviewTypes(ISymbol typeOrMethodSymbol, ISymbol previewInterfaceConstraintSymbol)
+        {
+            var typeParameters = typeOrMethodSymbol switch
+            {
+                INamedTypeSymbol typeSymbol => typeSymbol.TypeParameters,
+                IMethodSymbol methodSymbol => methodSymbol.TypeParameters,
+                _ => throw new ArgumentException($"Unexpected symbol '{typeOrMethodSymbol.Kind}'")
+            };
+
+            foreach (var typeParameter in typeParameters)
+            {
+                foreach (var constraintType in typeParameter.ConstraintTypes)
+                {
+                    if (previewInterfaceConstraintSymbol.Equals(constraintType, SymbolEqualityComparer.Default /*DetectPreviewFeatureSymbolEqualityComparer*/))
```

Comment:
PR isn't ready to merge until I make sure symbol comparison will behave the way it should, potentially by introducing a new comparer.